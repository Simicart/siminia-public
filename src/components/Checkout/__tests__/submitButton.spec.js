import React from 'react';
import testRenderer from 'react-test-renderer';
import SubmitButton from '../submitButton';

test('renders a SubmitButton component', () => {
    const props = {
        submitOrder: jest.fn(),
        submitting: false,
        valid: false
    };
    const component = testRenderer.create(<SubmitButton {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('sets disabled true if submitting', () => {
    const props = {
        submitOrder: jest.fn(),
        submitting: true,
        valid: false
    };
    const component = testRenderer.create(<SubmitButton {...props} />);
    expect(component.toJSON().props.disabled).toBe(true);
});

test('sets disabled true if not submitting and not valid', () => {
    const props = {
        submitOrder: jest.fn(),
        submitting: false,
        valid: false
    };
    const component = testRenderer.create(<SubmitButton {...props} />);
    expect(component.toJSON().props.disabled).toBe(true);
});

test('sets disabled false if not submitting and valid', () => {
    const props = {
        submitOrder: jest.fn(),
        submitting: false,
        valid: true
    };
    const component = testRenderer.create(<SubmitButton {...props} />);
    expect(component.toJSON().props.disabled).toBe(false);
});
