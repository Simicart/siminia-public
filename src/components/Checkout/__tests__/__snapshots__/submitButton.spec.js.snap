// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a SubmitButton component 1`] = `
<button
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  <span>
    Confirm Order
  </span>
</button>
`;
